### PR TITLE
remove the Protector interface, introduce a PSK type

### PR DESCRIPTION
**Pull request by marten-seemann**
Tracking issue: https://github.com/libp2p/go-libp2p/issues/795.

### CODE REVIEW

File: pnet/protector.go [hunk 1]
```diff
@@ -1,15 +1,4 @@
...
-// libp2p. It is created by implementation and use by libp2p-conn to secure connections
-// so they can be only established with selected number of peers.
-type Protector interface {
-	// Wraps passed connection to protect it
-	Protect(net.Conn) (net.Conn, error)
-
-	// Returns key fingerprint that is safe to expose
-	Fingerprint() []byte
-}
+type PSK []byte
```
Comment (Stebalien) [inline pnet/protector.go]: Why not `type PSK [32]byte`?

Comment (Stebalien) [reply pnet/protector.go]: Also, docs.

Comment (Stebalien) [reply pnet/protector.go]: I'm fine leaving this as `[]byte`, but it needs docs.

Comment (marten-seemann) [reply pnet/protector.go]: Done.

**Comment by Stebalien** [comment]
Tests are failing due to breaking changes. That's fine.

**Pull request merged**